clang/lib/AST/DeclCXX.cpp

```diff
@@ -3492,10 +3492,13 @@ VarDecl *BindingDecl::getHoldingVar() const {
   return VD;
 }
 
-llvm::ArrayRef<Expr *> BindingDecl::getBindingPackExprs() const {
+llvm::ArrayRef<BindingDecl *> BindingDecl::getBindingPackDecls() const {
   assert(Binding && "expecting a pack expr");
-  auto *RP = cast<ResolvedUnexpandedPackExpr>(Binding);
-  return RP->getExprs();
+  auto *FP = cast<FunctionParmPackExpr>(Binding);
+  ValueDecl *const *First = FP->getNumExpansions() > 0 ? FP->begin() : nullptr;
+  assert((!First || isa<BindingDecl>(*First)) && "expecting a BindingDecl");
+  return llvm::ArrayRef<BindingDecl *>((BindingDecl *const *)First,
```

Comment:
What you could do here is `&cast<BindingDecl>(*First)`, plus wrap that in a const-cast.  This would make the assert unnecessary.

Reply:
I did have something like that to start, but then I think I found that `cast` does not return a reference to a pointer
so the ArrayRef constructor will catch that.

Reply:
It says you marked this as resolved.

Reply:
This I did with the use of `cast`, which you gave some good reasons to skip.  But I separately commented that I want the C cast gone.